File: python/src/problem/leetcode/easy/leetcode_27.py
# coding=utf-8
from typing import List


class Solution:
    """
    移除数组中指定数字
    """

    def remove_element(self, nums: List[int], val: int) -> int:
        """
        Time: O(n), Space: O(1)
        :param nums:
        :param val:
        :return:
        """
        if not nums:
            return 0

        left = right = 0
        while right < len(nums):
            if nums[right] != val:
                nums[left] = nums[right]
                left += 1
            right += 1
        return left

File: java/src/main/java/com/ywh/problem/leetcode/easy/LeetCode27.java
package com.ywh.problem.leetcode.easy;

/**
 * 移除数组中指定数字
 * [双指针] [数组]
 *
 * @author ywh
 * @since 03/04/2020
 */
public class LeetCode27 {

    /**
     * Time: O(n), Space: O(1)
     *
     * @param nums
     * @param val
     * @return
     */
    public int removeElement(int[] nums, int val) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        int left = 0, right = 0;
        while (right < nums.length) {
            if (nums[right] != val) {
                nums[left++] = nums[right];
            }
            right++;
        }
        return left;
    }

}
